samples/24.customization-smart-display/README.md

```diff
@@ -0,0 +1,468 @@
+
+```jsx
+import React from 'react';
+
+import { Components, Constants } from 'botframework-webchat';
+
+const { connectDictationInterims } = Components;
+const {
+   DictateState: { DICTATING, STARTING }
+} = Constants;
+
+export default connectDictationInterims()(
+   ({ className, dictateInterims, dictateState }) =>
+      (dictateState === STARTING || dictateState === DICTATING) &&
+      !!dictateInterims.length && (
+         <p className={className}>
+            {dictateInterims.map((interim, index) => (
+               <span key={index}>{interim}&nbsp;</span>
+            ))}
+         </p>
+      )
+);
+```
+
+Next is the `LastBotActivity`, which will be displayed in place of a transcript.
+
+1. Import `React` and `Components` and `connectToWebChat` from `botframework-webchat`
+1. Extract `SpeakActivity` from `Components`
+1. Modify the function `connectToWebChat`, which will show the latest activity text from the bot as well as speak it.
+
+```jsx
+import React from 'react';
+
+import { connectToWebChat, Components } from 'botframework-webchat';
+
+const { SpeakActivity } = Components;
+
+export default connectToWebChat(({ activities }) => ({
+   activity: activities
+      .slice()
+      .reverse()
+      .find(({ from: { role }, type }) => role === 'bot' && type === 'message')
+}))(
+   ({ activity }) =>
+      !!activity && (
+         <React.Fragment>
+            <p>{activity.text}</p>
+            {activity.channelData && activity.channelData.speak && <SpeakActivity activity={activity} />}
+         </React.Fragment>
+      )
+);
+```
+
+Finally, let's return to `App.js`. Your imports should look like the following:
+
+```jsx
+import './App.css';
+import {
+   Components,
+   createDirectLine,
+   createCognitiveServicesSpeechServicesPonyfillFactory
+} from 'botframework-webchat';
+import React, { Component } from 'react';
+
+import CustomDictationInterims from './CustomDictationInterims';
+import CustomMicrophoneButton from './CustomMicrophoneButton';
+import fetchSpeechServicesToken from './fetchSpeechServicesToken';
+import LastBotActivity from './LastBotActivity';
+
+const { Composer } = Components;
+```
+
+Create and export the React App Component. Then set up the constructor, which will initialize the `state` of `directLine` as `null`
+
+```js
+export default class App extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      directLine: null
+    };
+  }
+…
+```
+
+We want to use the `componentDidMount()` lifecycle method to handle fetching a token from Mock Bot. It is **never recommended** to put the Direct Line secret in the browser or client app. To learn more about secrets and tokens for Direct Line, visit this [tutorial on authentication](https://docs.microsoft.com/en-us/azure/bot-service/rest-api/bot-framework-rest-direct-line-3-0-authentication).
+
```

Comment:
I stopped reviewing this file at line `254`. I think this file will need a lot of cleanup once we decide where to place it. I still think it belongs as 13.b, with more cross-referencing between 13a and 13b.